src/Database/LSMTree/Internal/IndexOrdinary.hs

```diff
@@ -72,15 +72,19 @@ search key (IndexOrdinary lastKeys)
 
     start :: Int
     start | protoStart < pageCount = protoStart
-          | otherwise              = maybe 0 succ $
+          | otherwise              = maybe 0 succ                              $
                                      findIndexR (/= lastKeys ! pred pageCount) $
                                      lastKeys
 
     end :: Int
-    end | protoStart < pageCount
-            = start + length (takeWhile (== key) (drop (succ start) lastKeys))
-        | otherwise
-            = pred pageCount
+    end | protoStart < pageCount = start + overflowPageCount
```

Comment:
There is the same guard in both of these definitions. Maybe we should put them under a single guard, returning `(Int, Int)` or even the `PageSpan`?

Reply:
Done as part of 9486ca27fdb2f335729f5f6f1d50982022e306ec.